### PR TITLE
Many changes

### PR DESCRIPTION
**Pull request by drivera-armedia**
Finalized support for privileged mode COW (used to capture the artifact caches)
Finalized support for disabling tools via .ignore files

### CODE REVIEW

**Review by naveen-armedia — APPROVED** [review]

**Review by naveen-armedia — DISMISSED** [review]

File: .gitignore [hunk 1]
```diff
@@ -1 +1,2 @@
 **/*.swp
+tools
```
Comment (david-oc-miller) [inline .gitignore:2]: we don't want to check the tools into git? How will we manage them, or back them up, or track changes?

File: Dockerfile [hunk 2]
```diff
@@ -27,37 +29,117 @@ LABEL IMAGE_SOURCE="https://github.com/ArkCase/ark_jenkins_build"
...
+COPY --chown=root:root 00-build /etc/sudoers.d
+RUN chmod 0640 /etc/sudoers.d/00-build
+
+#
+# Now do the configurations for the actual user
+#
+USER "${APP_USER}"
+
+#
+# Disable SSL certificate verification for GIT (for now ... this probably needs to go away)
```
Comment (david-oc-miller) [inline Dockerfile]: Right, we could use openssl to lift the root certificate from gitlab.armedia.com and add it to the image, so it becomes a trusted site.

**Review by david-oc-miller — DISMISSED** [review]

**Comment by david-oc-miller** [comment]
looks like another developer workshop...

File: Dockerfile (hunk 2 above)
Comment (drivera-armedia) [reply Dockerfile]: Yeah I'm moving this to a script in init.d so we can turn it on/off without messing with the image

File: .gitignore (hunk 1 above)
Comment (drivera-armedia) [reply .gitignore:2]: No. Specifically not. What we want is to be able to mess with the tool set without having to mess with the images. One of our current pain points has to do with how we're having to re-do the image every time a newer version of a build tool is required, which means that each image is tailored to a specific (set of) version(s) of our build tool(s).

By keeping this separate, our image is simplified significantly, and it gives us the ability to maintain those tools and configure them, etc., without requiring the baking of new build images.

Comment (david-oc-miller) [reply .gitignore:2]: How do we manage the tools folder?

Comment (david-oc-miller) [reply .gitignore:2]: Also I don't believe the .gitignore keeps files from being added to the image itself, it only keeps files out of git, unless there's some integration in the image build process to make the docker build consider .gitignore... maybe I'm missing something.

**Review by david-oc-miller — APPROVED** [review]

**Pull request merged**